internal/parse/read.go

```diff
@@ -23,33 +21,15 @@ func readLineByLine(scanner *bufio.Scanner, returnChannel chan<- []string, fnApp
 	returnChannel <- lines
 }
 
-func ScanLines(arg *Options, algo SortAlgorithm) (lines []string, err error) {
+func ScanLines(files []io.Reader, algo SortAlgorithm) (lines []string, err error) {
 	var scanner *bufio.Scanner
 	// using 1 channel for each file
-	var channel chan []string
-	if READ_STDIN {
-		channel = make(chan []string, 1)
-		scanner = bufio.NewScanner(os.Stdin)
+	channel := make(chan []string, len(files))
+	for _, f := range files {
+		scanner = bufio.NewScanner(f)
```

Comment:
This is also nicely done, simplifies the structure, and the code is not duplicated for stdin and files